### ISSUE TITLE
RCS1174 when used with System.Linq.Async

### ISSUE THREAD

**Issue by tomasKosar** [dotnet/roslynator#1615]
I have rule  RCS1174 enabled but when it is used with System.Linq.Async then I think it creates False Positive.

I am using this third party library
https://github.com/dotnet/reactive?tab=readme-ov-file#linq-for-iasyncenumerable-systemlinqasync
Not part of official .NET but latest release has +100M downloads.
https://www.nuget.org/packages/System.Linq.Async
Unfortunately they do not have much documentation as its community driven.

The example to reproduce could be this

```
    private async Task<bool> IsNotEmptyAsync(string text)
    {
        await Task.Delay(1);

        return !string.IsNullOrEmpty(text);
    }

    private void Execute()
    {
        IAsyncEnumerable<string> texts = new List<string>() { "asd", "qwe", string.Empty }.ToAsyncEnumerable();

        // here RCS1174 suggests to remove async/await but that is not possible as it then creates an error CS0029 and CS1662
        var notEmptyTexts = texts.WhereAwait(async t => await this.IsNotEmptyAsync(t));

        Console.WriteLine(string.Join(", ", notEmptyTexts.ToEnumerable()));
    }
```

I can only suppress the rule for this case or disable it altogether.

Or am I missing something?
